ui/listeners.go

```diff
@@ -104,5 +122,91 @@ func (mp *mainPage) OnSyncCanceled(willRestart bool) {
 		Stage: wallet.SyncCanceled,
 	}
 }
-func (mp *mainPage) OnSyncEndedWithError(err error)          {}
+
+func (mp *mainPage) OnSyncEndedWithError(err error) {}
+
 func (mp *mainPage) Debug(debugInfo *dcrlibwallet.DebugInfo) {}
+
+func (mp *mainPage) desktopNotifier(notifier interface{}) {
+	var notification string
+	switch t := notifier.(type) {
+	case dcrlibwallet.Transaction:
+		// remove trailing zeros from amount and convert to string
+		amount := strconv.FormatFloat(dcrlibwallet.AmountCoin(t.Amount), 'f', -1, 64)
+
+		defaultNotification := "Transaction notification of %s DCR"
+		notificationText := "You have %s %s DCR "
+
+		wallet := mp.pageCommon.multiWallet.WalletWithID(t.WalletID)
+		if wallet == nil {
+			notification = fmt.Sprintf(defaultNotification, amount)
+			return
+		}
+
+		getAccount := func(acct int32) string {
+			var account string
+			accountName, err := wallet.AccountName(acct)
+			if err != nil {
+				log.Error(err)
+			} else {
+				account = accountName
+			}
+			return account
+		}
+
+		// get source account
+		var txSourceAccount string
+		if t.Direction == dcrlibwallet.TxDirectionSent ||
+			t.Direction == dcrlibwallet.TxDirectionTransferred {
+			for _, input := range t.Inputs {
+				if input.AccountNumber != -1 {
+					txSourceAccount = getAccount(input.AccountNumber)
+				}
+			}
+		}
+
+		//get distination account
+		var txDestAccount, txDestinationAddress string
+		if t.Direction == dcrlibwallet.TxDirectionTransferred ||
+			t.Direction == dcrlibwallet.TxDirectionReceived ||
+			t.Direction == dcrlibwallet.TxDirectionSent {
+			for _, output := range t.Outputs {
+				if output.AccountNumber != -1 {
+					txDestAccount = getAccount(output.AccountNumber)
+				}
+				if output.AccountNumber == -1 {
+					txDestinationAddress = output.Address
+				}
+			}
+		}
+
+		switch {
+		case t.Direction == dcrlibwallet.TxDirectionReceived:
+			notification = fmt.Sprintf(notificationText+"to %s account in %s wallet.", "received", amount, txDestAccount, wallet.Name)
+		case t.Direction == dcrlibwallet.TxDirectionSent:
```

Comment:
Remove options  for sent and transferred  transactions and add options for stake transactions.